src/frontend/src/utils/with_options.rs

```diff
@@ -40,6 +42,7 @@ mod options {
 #[derive(Default, Clone, Debug, PartialEq, Eq, Hash)]
 pub struct WithOptions {
     inner: BTreeMap<String, String>,
+    secret_ref: BTreeMap<String, SecretRef>,
```

Comment:
I think the best way should be like

```rust
enum WithOptionValue {
    Plain(String),
    Secret(SecretRef),
}
inner: BTreeMap<String, WithOptionValue>
```

This natually reflects the idea that an option can be either a plaintext or an secret.

Meanwhile, we may still keep them as two fields in proto message, in order to keep backward compatibility.

Reply:
Yes I agree it might be a better way for the `WithOption` struct. The only problem it we are using <String, String> everywhere and it will cause a lot of code changes. We can refactor it later.